homeassistant/components/trello/config_flow.py

```diff
@@ -0,0 +1,118 @@
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Prompt user for Trello API credentials."""
+        if user_input is None:
+            return await self._show_creds_form()
+
+        self.api_key = user_input[CONF_API_KEY]
+        self.api_token = user_input[CONF_API_TOKEN]
+        self.trello_adapter = TrelloAdapter.from_creds(
+            user_input[CONF_API_KEY], user_input[CONF_API_TOKEN]
+        )
+
+        try:
+            member = await self._get_current_member()
+        except Unauthorized as ex:
+            return await self._show_error_creds_form(ex)
+
+        self.user_id = member.id
+        self.user_email = member.email
+        self.ids_boards = await self._fetch_all_boards()
+
+        return await self._show_board_form(self.ids_boards)
+
+    async def async_step_boards(self, user_input: dict[str, Any]) -> FlowResult:
+        """Select desired boards to have card counts of per list.
+
+        :param user_input: User's selected boards
+        """
+        board_ids = user_input[CONF_BOARD_IDS]
+
+        await self.async_set_unique_id(self.user_id)
+        self._abort_if_unique_id_configured()
+
+        config_data: dict[str, str] = self._get_config_data()
+        config_options = {CONF_BOARD_IDS: board_ids}
+
+        return self.async_create_entry(
+            title=self.user_email, data=config_data, options=config_options
+        )
+
+    def _get_config_data(self) -> dict[str, str]:
+        return {
+            CONF_API_KEY: self.api_key,
+            CONF_API_TOKEN: self.api_token,
+            CONF_USER_ID: self.user_id,
+            CONF_USER_EMAIL: self.user_email,
+        }
+
+    async def _show_creds_form(self) -> FlowResult:
+        return self.async_show_form(
+            step_id="user", data_schema=CREDS_FORM_SCHEMA, last_step=False
+        )
+
+    async def _show_board_form(
+        self, ids_boards: dict[str, dict[str, str]]
+    ) -> FlowResult:
+        return self.async_show_form(
+            step_id="boards",
+            data_schema=_get_board_select_schema(ids_boards),
+        )
+
+    async def _show_error_creds_form(self, ex: Unauthorized) -> FlowResult:
+        LOGGER.error("Unauthorized: %s)", ex)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=CREDS_FORM_SCHEMA,
+            errors={"base": "invalid_auth"},
+            last_step=False,
+        )
+
+    async def _fetch_all_boards(self) -> dict[str, dict[str, str]]:
+        return await self.hass.async_add_executor_job(self.trello_adapter.get_boards)
+
+    async def _get_current_member(self) -> Member:
+        return await self.hass.async_add_executor_job(self.trello_adapter.get_member)
+
+
+def _get_board_select_schema(boards: dict[str, dict]) -> Schema:
+    options = {key: value["name"] for key, value in boards.items()}
+    return vol.Schema({vol.Required(CONF_BOARD_IDS): cv.multi_select(options)})
```

Comment:
```suggestion
USER_FORM_SCHEMA = vol.Schema(
    {
        vol.Required(CONF_API_KEY): str,
        vol.Required(CONF_API_TOKEN): str,
    }
)


class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
    """Handle a config flow for the initial setup of a Trello integration."""

    VERSION = 1

    def __init__(self) -> None:
        """Initialize config flow."""
        self._data = {}
        self._trello_adapter: TrelloAdapter
        self._ids_boards: dict[str, dict[str, str]] = {}


    async def async_step_user(
        self, user_input: dict[str, Any] | None = None
    ) -> FlowResult:
        """Prompt user for Trello API credentials."""
        errors = {}
        if user_input:
            self._trello_adapter = TrelloAdapter.from_data(user_input)
            try:
                member = await self.hass.async_add_executor_job(self._trello_adapter.get_member)
            except Unauthorized as ex:
                LOGGER.exception("Unauthorized", ex)
                errors={"base": "invalid_auth"}
            else:
                self._data = {
                    **user_input,
                    CONF_USER_ID: member.id,
                    CONF_USER_EMAIL: member.email,
                }

                await self.async_set_unique_id(member.id)
                self._abort_if_unique_id_configured()

                self._ids_boards = await self.hass.async_add_executor_job(self._trello_adapter.get_boards)
                return await self.async_step_boards()

        return self.async_show_form(
            step_id="user", data_schema=USER_FORM_SCHEMA,errors=errors, last_step=False
        )

    async def async_step_boards(self, user_input: dict[str, Any]) -> FlowResult:
        """Select desired boards to have card counts of per list.

        :param user_input: User's selected boards
        """
        if user_input:
            return self.async_create_entry(
                title=self.user_email, data=self._data, options=user_input
            )

        options = {key: value["name"] for key, value in self._ids_boards.items()}
        return self.async_show_form(
            step_id="boards",
            data_schema=vol.Schema({vol.Required(CONF_BOARD_IDS): cv.multi_select(options)}),
        )
```

I cleaned up your config flow, and it is imo more readable.
Please point me to the location where `CONF_USER_ID` and `CONF_USER_EMAIL` is used in the integration.